discovery/support/chaincode/support.go

```diff
@@ -35,20 +35,58 @@ func NewDiscoverySupport(ci MetadataRetriever) *DiscoverySupport {
 	return s
 }
 
-func (s *DiscoverySupport) PoliciesByChaincode(channel string, cc string, _ ...string) []policies.InquireablePolicy {
-	chaincodeData := s.ci.Metadata(channel, cc, false)
+func (s *DiscoverySupport) PoliciesByChaincode(channel string, cc string, collections ...string) []policies.InquireablePolicy {
+	chaincodeData := s.ci.Metadata(channel, cc, collections...)
 	if chaincodeData == nil {
-		logger.Info("Chaincode", cc, "wasn't found")
+		logger.Infof("Chaincode %s wasn't found", cc)
 		return nil
 	}
+
+	// chaincode policy
 	pol := &common2.SignaturePolicyEnvelope{}
 	if err := proto.Unmarshal(chaincodeData.Policy, pol); err != nil {
-		logger.Warning("Failed unmarshaling policy for chaincode", cc, ":", err)
+		logger.Warningf("Failed unmarshaling policy for chaincode '%s': %s", cc, err)
 		return nil
 	}
 	if len(pol.Identities) == 0 || pol.Rule == nil {
-		logger.Warningf("Invalid policy, either Identities(%v) or Rule(%v) are empty:", pol.Identities, pol.Rule)
+		logger.Warningf("Invalid policy, either Identities(%v) or Rule(%v) are empty", pol.Identities, pol.Rule)
 		return nil
 	}
-	return []policies.InquireablePolicy{inquire.NewInquireableSignaturePolicy(pol)}
+	chaincodePolicy := inquire.NewInquireableSignaturePolicy(pol)
+
+	inquireablePolicies := map[policies.InquireablePolicy]bool{}
```

Comment:
1) You can do a map from **Foo** to **struct{}**, it's the common pattern for a set in Go.
2) How does this even work? **policies.InquireablePolicy** is an interface. How does Golang know to find it again in the map if it already exists? Look at an example I made [here](https://play.golang.org/p/n2PF37yKT6D).

I think that what we really need is to make the keys of the map, the byte representation of the policies and not the derived interfaces.

Reply:
I changed it to `make(map[string]struct{})` and set the keys to the string representation of the policy bytes (since byte slices themselves can't be keys)